controllers/resource/upserter.go

```diff
@@ -128,25 +129,36 @@ func (u *upserter) upsertDoguExposedServices(ctx context.Context, doguResource *
 	return collectedErrs
 }
 
-func (u *upserter) upsertDoguPVC(ctx context.Context, doguResource *k8sv1.Dogu) error {
-	newPVC, err := u.generator.CreateDoguPVC(doguResource)
-	if err != nil {
-		return fmt.Errorf("failed to generate pvc: %w", err)
-	}
+func (u *upserter) upsertDoguPVCs(ctx context.Context, doguResource *k8sv1.Dogu, dogu *core.Dogu) error {
+	if len(dogu.Volumes) > 0 {
+		newPVC, err := u.generator.CreateDoguPVC(doguResource)
+		if err != nil {
+			return fmt.Errorf("failed to generate pvc: %w", err)
+		}
 
-	existingPVC := &v1.PersistentVolumeClaim{}
+		existingPVC := &v1.PersistentVolumeClaim{}
 
-	err = u.client.Get(ctx, doguResource.GetObjectKey(), existingPVC)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-
-	// Only create a PVC but do not update it (for now) because updating immutable PVCs is tough
-	if apierrors.IsNotFound(err) {
-		err = u.updateOrInsert(ctx, doguResource.GetObjectKey(), &v1.PersistentVolumeClaim{}, newPVC, &longhornPVCValidator{})
-		if err != nil {
+		err = u.client.Get(ctx, doguResource.GetObjectKey(), existingPVC)
+		if err != nil && !apierrors.IsNotFound(err) {
 			return err
 		}
+
+		objectKey := client.ObjectKey{Name: newPVC.Name, Namespace: newPVC.Namespace}
+		// Only create a PVC but do not update it (for now) because updating immutable PVCs is tough
+		if apierrors.IsNotFound(err) {
+			err = u.updateOrInsert(ctx, objectKey, &v1.PersistentVolumeClaim{}, newPVC, &longhornPVCValidator{})
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	newReservedPVC, err := u.generator.CreateReservedPVC(doguResource)
```

Comment:
🚫 **[checkstyle]** <ineffassign> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>ineffectual assignment to err